### ISSUE TITLE
Announce empty results in pickers each time picker value changes

### ISSUE THREAD

**Issue by miguelsolorio** [microsoft/vscode#94188]
Refs #93648

![image](https://user-images.githubusercontent.com/35271042/78076770-e76b2b80-735b-11ea-82b0-ea44e896a359.png)

When on the normal quick pick, the screen reader will say "0 results". But in the recent quick pick there's no mention of it. These should be consistent.

**Comment by misolori** [comment]
(sorry for the late post, I had written this one yesterday but never posted it 😞)

**Comment by bpasero** [comment]
To investigate if this is a regression or just flaky. 

@misolori was this testing with VoiceOver?

**Comment by misolori** [comment]
I used the [nvda](https://www.nvaccess.org/download/) voice over on Windows

**Comment by bpasero** [comment]
I cannot reproduce this, I get to 0 results when typing enough:

![image](https://user-images.githubusercontent.com/900690/78220201-5dd56e00-74c1-11ea-88f0-fe54f15d167b.png)

also for recently opened:

![image](https://user-images.githubusercontent.com/900690/78220287-85c4d180-74c1-11ea-822d-1672d5d630e6.png)

**Comment by jvesouza** [comment]
I tried the following steps using orca:
1. I pressed the f1 key.
2. I typed abcd.

Orca announced 0 results.

Sorry for my ignorance, but it is necessary to configure code-insiders to use recent quick pick?
Thanks.

**Comment by bpasero** [comment]
@jvesouza no, insiders will have the new quick pick enabled by default now without settings required

**Comment by jvesouza** [comment]
I did tests with other quick open commands and apparently I had no problem.
What caught my attention was the quick open view command that seems to behave differently from the others.
When I tried to use this command, I was unable to type anything to select the desired view.

**Comment by bpasero** [comment]
@jvesouza I appreciate you testing them, can you elaborate a bit more how for me to reproduce the issue you see with the view picker?

**Comment by jvesouza** [comment]
Steps to reproduce:
1. Press f1
2. Type quick open view
3. Press enter.

The orca just announced list box.

4. Type something.

The orca just announces the typed characters, nothing more.

**Comment by bpasero** [comment]
I think https://github.com/microsoft/vscode/issues/93610 would fix that issue. We need to select the first item once you start searching.

**Comment by misolori** [comment]
@bpasero it's when you have zero results and you keep typing:

![image](https://user-images.githubusercontent.com/35271042/78259211-3aa2c280-74b1-11ea-8bc9-4046bf732699.png)

I would expect for the "0 results" to keep being read out loud as we already do on every keystroke.

**Comment by misolori** [comment]
@bpasero it's when you have zero results and you keep typing:

![image](https://user-images.githubusercontent.com/35271042/78259211-3aa2c280-74b1-11ea-8bc9-4046bf732699.png)

I would expect for the "0 results" to keep being read out loud as we already do on every keystroke.

**Comment by bpasero** [comment]
After my fix to show an entry for some pickers at least you will now see this when continuing to type and there are no results:

![image](https://user-images.githubusercontent.com/900690/78265238-76b34300-7504-11ea-9610-7b53a1e97635.png)

But this is not applying to pickers that do not provide this result. 

I am not 100% sure what the right solution is here, e.g. would someone with a screen reader expect to get the "0 results" everytime, or is it fine when transitioning back to actually getting results.

**Comment by misolori** [comment]
I think the issue I ran into was as I was typing a long name, I partially heard the "0 results" but kept typing because I wasn't finished and then when i was done I didn't hear anything and thought that was odd.

**Comment by jvesouza** [comment]
For me the ideal is that we always hear 0 results while typing.
Sometimes, when we type very quickly, the announcement of the character typed by the orca interrupts the announcement of the results.
With that, we can miss the result announcement if it is done only once.
I don't know if I can explain it clearly.

**Comment by misolori** [comment]
@jvesouza that's exactly the issue I ran into, the announcement of the results was interrupted by typing another character. and if you were already at 0 results, subsequent additions will announce nothing.

**Comment by bpasero** [comment]
I wonder if we should just always have a generic "No results found" pick in all pickers when no result is found. This seems to work very well for those pickers that have it.

**Comment by isidorn** [comment]
A generic "No results found" makes sense for screen readers as well as a general ux solution. Also I think it aligns well with Search viewlet / references viewlet.

**Comment by chrmarti** [comment]
Not sure if that's still an option UX-wise, but we could try to get '0 Result' announced after each keystroke. (Maybe by using two `aria-live` elements.)

If we always show a "No results found" entry, should we also show a "No items supplied" entry when the picker was opened with no items at all? (Suboptimal maybe, but there are too many extensions to make this consistent without a built-in solution.)

**Comment by jvesouza** [comment]
It seems that I encountered a small problem in the current implementation using orca.
I'm not sure if it happens to any other screen reader.
Steps:
1. Press f1.
2. type slowly the following:
lamad

After I type the letter d, orca announces the following:
no matching commands, one results.

**Comment by chrmarti** [comment]
@jvesouza That's because 'No Matching Commands' is an entry in the list itself. When you keep typing, does it repeat that there are no matching commands, one result? I don't get that with VoiceOver and not getting that was the original problem we are trying to fix, if I understand correctly.

**Comment by jvesouza** [comment]
> @jvesouza That's because 'No Matching Commands' is an entry in the list itself. When you keep typing, does it repeat that there are no matching commands, one result? I don't get that with VoiceOver and not getting that was the original problem we are trying to fix, if I understand correctly.
No, I only hear it once.

**Comment by chrmarti** [comment]
Tried having two counts and alternate between the two to get the number repeated, but at least VoiceOver detects the duplication and does not reread the same number.

**Comment by isidorn** [comment]
@chrmarti fyi it should be pretty easy to setup Orca screen reader - if you have Ubuntu 18 it just worked for me. Here are the steps https://github.com/microsoft/vscode/issues/91063

**Comment by chrmarti** [comment]
I can reproduce with VoiceOver. Haven't found a way around it so far. From what I observe, this might be expected behavior on the VS Code side because the screen readers try to avoid reading the same information over and over. They also prioritize giving feedback about keypresses over reading out the live region and the combination of these two behaviors result in the suboptimal UX.

**Comment by isidorn** [comment]
Unassigning myself. @chrmarti let me know if you need me to investigate something or if I can help in any way here.

**Comment by bpasero** [comment]
I am not sure what to do here, e.g. in this case showing "No Results" as a pick seems odd:

![image](https://user-images.githubusercontent.com/900690/79536332-1b469080-8080-11ea-9bcf-9cd085061ecc.png)

Arguably it is already a bit weird that you can narrow down the actions until none appear because this is more like a menu with choices that should probably not get filtered at all. 

I think the "No results" message can probably not be added to all pickers as it depends on the use case.

**Comment by misolori** [comment]
I agree, I would not consider this a "no results" pick. I wonder if we can detect when there are options and then decide when we announce this.

**Comment by TylerLeonhardt** [comment]
This is actually fixed now, the recent quick pick says "0 results" with NVDA.
